pkg/infoprovider/rdmaInfoProvider.go

```diff
@@ -47,11 +54,36 @@ func (ip *rdmaInfoProvider) GetDeviceSpecs() []*pluginapi.DeviceSpec {
 		glog.Errorf("GetDeviceSpecs(): rdma is required in the configuration but the device is not rdma device")
 		return nil
 	}
-	return ip.rdmaSpec.GetRdmaDeviceSpec()
+
+	devsSpec := ip.rdmaSpec.GetRdmaDeviceSpec()
+	for _, devSpec := range devsSpec {
```

Comment:
I didn't want to run the function again it does a lot file reading and cutting strings I don't think there is a reason to run this function 2 times.

another option I can do is to save the devsSpec and run on them in the getEnv but running the GetRdmaDeviceSpec twice sounds like a bit of duplication/performance

Reply:
GetRdmaDeviceSpec() is just a getter, it has no special logic. am i missing anything ?

you would call it twice but you would not depend on the order of things.
ATM if someone calls GetEnvVal but did not call GetDeviceSpec, it would get an empty result

Reply:
no problem I will use it also in the get env